### PR TITLE
Feature/metrics by category 2

### PR DESCRIPTION
**Pull request by ajain-ida**
Allow metrics_by_category for MAE-273

### CODE REVIEW

File: fife/__main__.py [hunk 1]
```diff
@@ -74,11 +82,33 @@ def main():
...
+
+        if by_feature is not None:
+            values = list(set(data[by_feature]))
+
+            for feature_value in values:
+
+                evaluation_subset_by_period = modeler.data["_period"] == (
+                    modeler.data["_period"].max()
+                    - min(test_intervals, max_test_intervals)
+                )
```
Comment (jamesmckownbishop) [inline fife/__main__.py:98]: If we moved lines 100-102 above line 90, could we avoid subsetting with each iteration of the for loop?

Comment (ajain-ida) [reply fife/__main__.py:98]: Yes! Good catch, I didn't even think about that! I made the change, am running build packages and then will commit it to the current branch

File: fife/__main__.py [hunk 2]
```diff
@@ -52,6 +52,14 @@ def main():
     )
     print(f"Data processing time: {time() - checkpoint_time} seconds")
 
+    by_feature = config.get("BY_FEATURE", None)
+    if by_feature == "None":
+        by_feature = None
+    if by_feature is not None and by_feature not in data.columns:
```
Comment (jamesmckownbishop) [inline fife/__main__.py:58]: I like that you added some error handling here

File: fife/__main__.py [hunk 3]
```diff
@@ -52,6 +52,14 @@ def main():
     )
     print(f"Data processing time: {time() - checkpoint_time} seconds")
 
+    by_feature = config.get("BY_FEATURE", None)
+    if by_feature == "None":
```
Comment (jamesmckownbishop) [inline fife/__main__.py:56]: "None" is a reasonable default value, but for conformity with the rest of the API let's use the empty string "" as the default value instead.

**Pull request merged**